### PR TITLE
Misc small API additions to System.Reflection.Metadata

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera

**Comment by tmat** [comment]
CentOS failure is unrelated to the change: #5660

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/HandleComparer.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public int GetHashCode(EntityHandle obj)
         /// Compares two handles.
         /// </summary>
         /// <remarks>
-        /// The order of handles that differ in kind is undefined.
+        /// The order of handles that differ in kind and are not <see cref="EntityHandle"/> is undefined.
```
Comment (nguerrera) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/HandleComparer.cs]: Perhaps we should be more specific about what the defined order is.

**Comment by nguerrera** [comment]
LGTM.

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/HandleComparer.cs (hunk 1 above)
Comment (tmat) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/HandleComparer.cs]: Maybe. I'll leave that for another day though...

**Pull request merged**